### ISSUE TITLE
Unstable endwise

### ISSUE THREAD

**Issue by fuelen** [windwp/nvim-autopairs#74]
lua - works only for nested clauses and only from the second attempt:
[![asciicast](https://asciinema.org/a/v7Ovd1mCaMSDNccVb3I2De7et.svg)](https://asciinema.org/a/v7Ovd1mCaMSDNccVb3I2De7et)
elixir - always works only from the second attempt :
[![asciicast](https://asciinema.org/a/lRve8sd89EcZHtwhz8OGOc3x8.svg)](https://asciinema.org/a/lRve8sd89EcZHtwhz8OGOc3x8)
my config
```lua
local npairs = require("nvim-autopairs")
npairs.setup()
require("nvim-autopairs.completion.compe").setup(
    {
        map_cr = true, --  map <CR> on insert mode
        map_complete = true -- it will auto insert `(` after select function or method item
    }
)

local endwise = require("nvim-autopairs.ts-rule").endwise
npairs.add_rules(require("nvim-autopairs.rules.endwise-lua"))

npairs.add_rules(
    {
        endwise(" do$", "end", "elixir", nil)
    }
)
```

**Comment by sbaildon** [comment]
I took your elixir configuration and tried it out—no problems for me. Maybe there was a change between then and now?

caveat: I did remove `npairs.add_rules(require("nvim-autopairs.rules.endwise-lua"))`

**Comment by fuelen** [comment]
@sbaildon I don't know what happened, but I just tried it again, and it works. Thank you for checking.

Okay, endwise works for Elixir, but there is still an issue with lua and nested blocks.

I've removed  `npairs.add_rules(require("nvim-autopairs.rules.endwise-lua"))` and defined a custom rule for lua without tree-sitter checking `endwise('then$', 'end', 'lua', nil)` and issue with nested blocks dissapeared.
